mlflow/pyfunc/utils/data_validation.py

```diff
@@ -57,9 +74,21 @@ def wrapper(*args, **kwargs):
     return wrapper
 
 
+def _check_func_signature(func, func_name) -> list[str]:
+    parameters = inspect.signature(func).parameters
+    param_names = list(filter(lambda x: x != "self", list(parameters.keys())))
+    if invalid_params := set(param_names) - {"context", "model_input", "params"}:
```

Comment:
```suggestion
    if invalid_params := set(param_names) - {"self", "context", "model_input", "params"}:
```

does this work?

Reply:
Oh but we still need param_names afterwards to get the input param name, so we still need to exclude "self" from the list